# Ames Housing Regression Project

## Test 1: Local Data Ingestion and Structural Verification
In this test, I intend to load the raw dataset from my local directory into a Pandas DataFrame. My objective is to verify that the file path is correct, ensure the data is loaded without corruption, and inspect the initial shape (total rows and columns) of the dataset.

In [13]:
import pandas as pd 
import os 

In [27]:
# 1. Define the relative path to the locally downloaded file
# 'data' is a folder in the same directory as this notebook
data_path = os.path.join("..","data", "raw", "train.csv")

# 2. Load the dataset using pandas
df_raw = pd.read_csv(data_path)

#Verify the loading success and display dimensions
print("[SUCCESS] Local dataset successfully loaded into df_raw!")
print(f"Dataset Shape: {df_raw.shape[0]} rows (houses) and {df_raw.shape[1]} columns (features).")

[SUCCESS] Local dataset successfully loaded into df_raw!
Dataset Shape: 1460 rows (houses) and 81 columns (features).


## 2. Preliminary Data Discovery
Statistical summary of the core business features to understand market prices, property sizes, and temporal distribution.

In [28]:
# 3. Select core features to understand the business layout
features_of_interest = ['SalePrice', 'LotArea', 'YearBuilt', 'FullBath', 'BedroomAbvGr']

# 4. Display clean, human-readable statistics without messy float decimals
summary_df = df_raw[features_of_interest].describe()
display(summary_df.style.format({
    'SalePrice': '${:,.2f}',
    'LotArea': '{:,.0f}',
    'YearBuilt': '{:.0f}',
    'FullBath': '{:.1f}',
    'BedroomAbvGr': '{:.1f}'
}))

,SalePrice,LotArea,YearBuilt,FullBath,BedroomAbvGr
count,"$1,460.00","1,460",1460,1460.0,1460.0
mean,"$180,921.20","10,517",1971,1.6,2.9
std,"$79,442.50","9,981",30,0.6,0.8
min,"$34,900.00","1,300",1872,0.0,0.0
25%,"$129,975.00","7,554",1954,1.0,2.0
50%,"$163,000.00","9,478",1973,2.0,3.0
75%,"$214,000.00","11,602",2000,2.0,3.0
max,"$755,000.00","215,245",2010,3.0,8.0


## 3. Test 1 Insights & Conclusions
* **Dataset Scope:** The training set contains `1,460` records across `81` features, presenting a high-dimensional regression problem.
* **Market Profile:** The average sale price is **$180,921.20**, ranging from entry-level homes at **$34,900.00** up to luxury estates at **$755,000.00**.
* **Temporal Span:** The data captures over a century of construction history, featuring properties built from **1872** up to **2010**.

> 📌 **Next Milestone (Test 2):** Investigate missing data structures and interpret null values (e.g., `PoolQC`, `FireplaceQu`) to plan the data cleaning strategy.